master/internal/kubernetes/request_queue_test.go

```diff
@@ -0,0 +1,191 @@
+package kubernetes
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"gotest.tools/assert"
+
+	petName "github.com/dustinkirkland/golang-petname"
+
+	"github.com/determined-ai/determined/master/pkg/actor"
+
+	k8sV1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakeK8 "k8s.io/client-go/kubernetes/fake"
+	typedV1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+type (
+	mockPodActorPing struct{}
+	deleteMockPod    struct{}
+)
+
+type mockPodActor struct {
+	requestQueue *actor.Ref
+	name         string
+}
+
+func newMockPodActor(requestQueue *actor.Ref) *mockPodActor {
+	return &mockPodActor{
+		requestQueue: requestQueue,
+		name:         petName.Generate(3, "-"),
+	}
+}
+
+func (m *mockPodActor) Receive(ctx *actor.Context) error {
+	switch msg := ctx.Message().(type) {
+	case actor.PreStart:
+		podSpec := k8sV1.Pod{ObjectMeta: metaV1.ObjectMeta{Name: m.name}}
+		cmSpec := k8sV1.ConfigMap{ObjectMeta: metaV1.ObjectMeta{Name: m.name}}
+
+		ctx.Tell(m.requestQueue, createKubernetesResources{
+			handler:       ctx.Self(),
+			podSpec:       &podSpec,
+			configMapSpec: &cmSpec,
+		})
+
+	case mockPodActorPing:
+		ctx.Respond(mockPodActorPing{})
+
+	case deleteMockPod:
+		ctx.Ask(m.requestQueue, deleteKubernetesResources{
+			handler:       ctx.Self(),
+			podName:       m.name,
+			configMapName: m.name,
+		})
+
+	case resourceCreationCancelled, resourceDeletionFailed:
+
+	default:
+		ctx.Log().Errorf("unexpected message %T", msg)
+		return actor.ErrUnexpectedMessage(ctx)
+	}
+
+	return nil
+}
+
+func getNumberOfActivePods(podInterface typedV1.PodInterface) int {
+	// Because the k8s fake client doesn't not always process requests
```

Comment:
question: we might hit this in k8s, right. Even after `create` and `delete` are finished, `list` might not return the updated result.

Reply:
Yeah its possible, but in our non-test code we don't rely on List providing accurate real time information, we process it as it comes in.